app/controllers/api/v1/accounts/pins_controller.rb

```diff
@@ -27,4 +34,66 @@ def set_account
   def relationships_presenter
     AccountRelationshipsPresenter.new([@account.id], current_user.account_id)
   end
+
+  def load_accounts
+    return [] if hide_results?
+
+    if unlimited?
+      endorsed_accounts.all
+    else
+      endorsed_accounts.paginate_by_max_id(
+        limit_param(limit),
+        params[:max_id],
+        params[:since_id]
+      )
+    end
+  end
+
+  def endorsed_accounts
+    @account.endorsed_accounts
+  end
+
+  def hide_results?
+    @account.suspended? || (current_account && @account.blocking?(current_account))
+  end
+
+  def insert_pagination_headers
+    set_pagination_headers(next_path, prev_path)
+  end
+
+  def next_path
+    return if unlimited?
+
+    api_v1_account_pinned_index_url pagination_params(max_id: pagination_max_id) if records_continue?
+  end
+
+  def prev_path
+    return if unlimited?
+
+    api_v1_account_pinned_index_url pagination_params(since_id: pagination_since_id) unless @accounts.empty?
+  end
+
+  def pagination_max_id
+    @accounts.last.id
+  end
+
+  def pagination_since_id
+    @accounts.first.id
+  end
+
+  def records_continue?
+    @accounts.size == limit_param(limit)
+  end
+
+  def pagination_params(core_params)
+    params.slice(:limit).permit(:limit).merge(core_params)
+  end
+
+  def limit
+    params[:limit].to_i
```

Comment:
Adjacent to https://github.com/polyamspace/mastodon/pull/298/files#r1398565619
This should fall back to `DEFAULT_ACCOUNTS_LIMIT